projects/storefrontapp-e2e-cypress/cypress/integration/regression/customer-ticketing/create-ticket.e2e-spec.ts

```diff
@@ -140,6 +147,49 @@ describe('ticketing', () => {
+          category: TestCategory.complaint,
+        };
+        customerTicketing.loginRegisteredUser();
+        customerTicketing.visitElectronicTicketListingPage();
+        customerTicketing.openCreateTicketPopup();
+        customerTicketing.fillTicketDetails(testTicketDetails);
+        customerTicketing.clickSubmit();
+        customerTicketing.verifyFieldInputDoesntExceedLimit();
+      });
+
+      it('should close create ticket popup upon submit', () => {
+        const testTicketDetails: TestTicketDetails = {
+          subject: 'Entering a subject',
+          message: 'Typing a message',
+          category: TestCategory.complaint,
+        };
+
+        customerTicketing.loginRegisteredUser();
+        customerTicketing.visitElectronicTicketListingPage();
+        customerTicketing.openCreateTicketPopup();
+        customerTicketing.fillTicketDetails(testTicketDetails);
+        customerTicketing.clickSubmit();
```

Comment:
nit:  these lines are repeated in your three added tests and they might be repeated in other tests:
How about extracting a function in the same file with these lines:
```
function createTicketJourneyWithDetails(ticketDetails: TestTicketDetails) {
        customerTicketing.loginRegisteredUser();
        customerTicketing.visitElectronicTicketListingPage();
        customerTicketing.openCreateTicketPopup();
        customerTicketing.fillTicketDetails(testTicketDetails);
        customerTicketing.clickSubmit();
}

Reply:
I had a question regarding this; would this not be confusing to read at the first time since we want to make sure that every function we call on the spec file to be specific?

Reply:
ok, leave it as it is